fdbserver/DataDistribution.actor.cpp

```diff
@@ -3796,6 +3880,255 @@ ACTOR Future<vector<std::pair<StorageServerInterface, ProcessClass>>> getServerL
+				watchFuture = res.first;
+				self->wigglingPid = Optional<Key>(res.second);
+				StringRef pid = self->wigglingPid.get();
+
+				if (self->healthyTeamCount <= 1) { // pre-check health status
+					pauseWiggle.trigger();
+				} else {
+					auto fv = self->excludeStorageServersForWiggle(pid);
+					movingCount = fv.size();
+					moveFinishFuture = waitForAll(fv);
+					TraceEvent("PerpetualStorageWiggleStart", self->distributorId)
+					    .detail("ProcessId", pid)
+					    .detail("StorageCount", movingCount);
+				}
+			}
+			when(wait(restart.onTrigger())) {
+				if (self->wigglingPid.present()) {
+					StringRef pid = self->wigglingPid.get();
+					auto fv = self->excludeStorageServersForWiggle(pid);
+					moveFinishFuture = waitForAll(fv);
+					TraceEvent("PerpetualStorageWiggleRestart", self->distributorId)
+					    .detail("ProcessId", pid)
+					    .detail("StorageCount", fv.size());
+					isPaused = false;
+				}
+			}
+			when(wait(moveFinishFuture)) {
+				ASSERT(self->wigglingPid.present());
+				StringRef pid = self->wigglingPid.get();
+
+				moveFinishFuture = Never();
+				self->includeStorageServersForWiggle(pid);
+				TraceEvent("PerpetualStorageWiggleFinish", self->distributorId)
+				    .detail("ProcessId", pid.toString())
+				    .detail("StorageCount", movingCount);
+
+				self->wigglingPid.reset();
+				finishStorageWiggleSignal.send(Void());
+			}
+			when(wait(self->zeroHealthyTeams->onChange())) {
+				if (self->zeroHealthyTeams->get() && !isPaused) {
+					pauseWiggle.trigger();
+				}
+			}
+			when(wait(ddQueueCheck)) { // check health status periodically
+				Promise<int> countp;
+				self->getUnhealthyRelocationCount.send(countp);
+				int count = wait(countp.getFuture());
+
+				if (count >= SERVER_KNOBS->DD_STORAGE_WIGGLE_PAUSE_THRESHOLD && !isPaused) {
+					pauseWiggle.trigger();
+				} else if (count < SERVER_KNOBS->DD_STORAGE_WIGGLE_PAUSE_THRESHOLD && self->healthyTeamCount > 1 &&
+				           isPaused) {
+					restart.trigger();
+				}
+				ddQueueCheck = delay(SERVER_KNOBS->DD_ZERO_HEALTHY_TEAM_DELAY, TaskPriority::DataDistributionLow);
+			}
+			when(wait(pauseWiggle.onTrigger())) {
+				if (self->wigglingPid.present()) {
+					StringRef pid = self->wigglingPid.get();
+					isPaused = true;
+					moveFinishFuture = Never();
+					self->includeStorageServersForWiggle(pid);
+					TraceEvent("PerpetualStorageWigglePause", self->distributorId)
+					    .detail("ProcessId", pid)
+					    .detail("StorageCount", movingCount);
+				}
+			}
+		}
+	}
+
+	if (self->wigglingPid.present()) {
+		self->includeStorageServersForWiggle(self->wigglingPid.get());
+		self->clearWigglingPidAfterRecruitment = true;
+	}
+
+	return Void();
+}
+
+// This coroutine sets a watch to monitor the value change of `perpetualStorageWiggleKey` which is controlled by command
+// `configure perpetual_storage_wiggle=$value` if the value is 0, this actor start 2 actors,
```

Comment:
```suggestion
// `configure perpetual_storage_wiggle=$value` if the value is 1, this actor start 2 actors,
```

Shouldn't it be one? Or otherwise I am a bit confused how this is working